File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxVal=0;
        int minVal=9999999;
        int maxProfit=0;
        int profit;
        for (int price:prices){
            minVal=Math.min(minVal,price);
            profit=price-minVal;
            maxProfit=Math.max(maxProfit,profit);
        }
        return maxProfit;
    }
}

File: Data Structures & Algorithms/house-robber/submission-4.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums)<=1:
            return nums[0]
        DP=[None]*len(nums)
        DP[0]=nums[0]
        DP[1]=max(nums[1],nums[0])
        for i in range(2,len(DP)):
            val=max(DP[i-1],DP[i-2]+nums[i])
            DP[i]=val
        return DP[-1]